src/main/java/org/elasticsearch/index/engine/InternalEngine.java

```diff
@@ -797,7 +786,7 @@ public void forceMerge(final boolean flush, int maxNumSegments, boolean onlyExpu
     }
 
     @Override
-    public SnapshotIndexCommit snapshotIndex() throws EngineException {
+    public SnapshotIndexCommit snapshotIndex(boolean flushFirst) throws EngineException {
```

Comment:
Can you add javadoc about what kind of flush happens in this method (non-translog, waiting for ongoing)